src/weights.jl

```diff
@@ -530,6 +623,15 @@ _mean(A::AbstractArray, w::AbstractWeights, dims::Nothing) =
 _mean(A::AbstractArray{T}, w::AbstractWeights{W}, dims::Int) where {T,W} =
     _mean!(similar(A, wmeantype(T, W), Base.reduced_indices(axes(A), dims)), A, w, dims)
 
+function StatsBase._mean(A::AbstractArray, w::UnitWeights{S,T}, dims::Nothing) where {S,T}
+    (length(A) != length(w)) && throw(DimensionMismatch("Inconsistent array dimension."))
+    return mean(A)
```

Comment:
I suspect you can drop this method an just pass `dims=dims` even when it's `nothing`.

Reply:
So did I, but it doesn't actually work.

Reply:
Too bad. Though it would be shorter to use a single function, with a condition `dims === nothing ? mean(A) : mean(A, dims=dims)`.

Reply:
It isn't much shorter, because the methods have different boundary checks (which needed fixing, incidentally), so I left them as they are, but I am open to changing it.

Reply:
Ah, sorry.